### PR TITLE
config-tools: support of passthrough dTPM2 

### PR DESCRIPTION
**Pull request by yuchuyang**
1. Parse and insert tpm2 acpi table information to board.xml
2. Allocate tpm2 log area address
3. Create tpm2 mmiodev in vm_configurations.c
4. Create virtual tpm2 acpi table

### CODE REVIEW

**Comment by yuchuyang** [comment]
@wenlingz  Please hold this PR. Other PRs need to be merged before this one.

**Comment by wenlingz** [comment]
> @wenlingz Please hold this PR. Other PRs need to be merged before this one.

OK

**Review by NanlinXie — APPROVED** [review]

File: misc/config_tools/xforms/vm_configurations.c.xsl [hunk 1]
```diff
@@ -238,18 +239,24 @@
...
       <xsl:text>},</xsl:text>
       <xsl:value-of select="$newline" />
       <xsl:value-of select="$endif" />
       <xsl:value-of select="acrn:ifdef('P2SB_BAR_ADDR')" />
       <xsl:value-of select="acrn:initializer('pt_p2sb_bar', 'true')" />
       <xsl:value-of select="acrn:initializer('mmiodevs[0]', '{', true())" />
-      <xsl:value-of select="acrn:initializer('user_vm_pa', 'P2SB_BAR_ADDR_GPA')" />
-      <xsl:value-of select="acrn:initializer('service_vm_pa', 'P2SB_BAR_ADDR')" />
-      <xsl:value-of select="acrn:initializer('size', 'P2SB_BAR_SIZE')" />
+      <xsl:value-of select="acrn:initializer('res[0].base_gpa', 'P2SB_BAR_ADDR_GPA')" />
```
Comment (lifeix) [inline misc/config_tools/xforms/vm_configurations.c.xsl:257]: please rename base_gpa to user_vm_pa and base_hpa to host_hpa

Comment (lifeix) [reply misc/config_tools/xforms/vm_configurations.c.xsl:257]: Also, please config CONFIG_SECURITY_VM_FIXUP for the platform which needs to do TPM2 fixup, and config the VM flag GUEST_FLAG_TPM2_FIXUP for the VM which needs to do TPM2 fixup.

**Comment by lifeix** [comment]
This patch set is integrated into https://github.com/projectacrn/acrn-hypervisor/pull/6367.

**Pull request closed without merging**